primitives/transaction/src/account/basic_account.rs

```diff
@@ -41,8 +41,14 @@ impl AccountTransactionVerification for BasicAccountVerifier {
         assert_eq!(transaction.sender_type, AccountType::Basic);
 
         // Verify signer & signature.
-        let signature_proof: SignatureProof =
-            Deserialize::deserialize_from_vec(&transaction.proof[..])?;
+        let signature_proof: SignatureProof = match transaction.format() {
+            TransactionFormat::Basic => SignatureProof::EdDSA(
+                EdDSASignatureProof::deserialize_from_vec(&transaction.proof[..])?,
```

Comment:
If we go through with these changes, I think we should go all the way and implement it for basic txns.

@jsdanielh @viquezclaudio What are the implications for changing the format of transactions w.r.t. the old history? It's not a new problem (i.e., we have a solution), right?